validator/project_validator.go

```diff
@@ -128,11 +125,19 @@ var projectSyntaxValidators = []projectValidator{
 }
 
 // Functions used to validate the semantics of a project configuration file.
-var projectSemanticValidators = []projectValidator{
+var projectWarningValidators = []projectValidator{
```

Comment:
@hadjri, is there ever a time we want just warnings? Right now to separate these validators out, we've had to create new validators that loop more times, which seems like an unnecessary performance hit if we do typically want 1) Only Errors and 2) Errors and Warnings.

Maybe we could have these functions instead:
CheckProjectErrors (runs all functions that return errors but filters out the warnings) and CheckProjectWarningsAndErrors (runs all functions). I don't think there's anywhere where we only want to return warnings and not errors anyway, but we could brainstorm other formats in that case, or stick with this one. (We can also discuss outside of this thread.)

Reply:
I don't think there is anywhere at the moment that explicitly only needs warnings - I think that sounds reasonable, since there are currently a few places where we only want to check for errors and having this organization would be cleaner.

Reply:
SGTM

Reply:
Oh my mistake. CheckProjectWarnings should run even when there are no errors. I have modified api.go so that the function will run even when there are no errors. The only time errors will not run is if input.Quiet is set to false. https://github.com/evergreen-ci/evergreen/blob/9320bb372bdb550a7175d201b8e4e1035e234f99/service/api.go#L670-L688